hoomd/filter/filter_.py

```diff
@@ -0,0 +1,14 @@
+class ParticleFilter:
```

Comment:
Are users supposed to subclass this? If not, make it private. In either case, you may want to make it an abstract class.

Reply:
We can't make it an abstract class (I tried). The way pybind11 deals with inheritance prevents using metaclasses here. We would have to switch to a composition model for this to work.